rust_src/src/eval_macros.rs

```diff
@@ -129,7 +129,95 @@ macro_rules! def_lisp_sym {
     ($name:expr, $value:expr) => {};
 }
 
-#[allow(unused_macros)]
+/// Macros we use to define forwarded Lisp variables.
+/// These are used in the syms_of_FILENAME functions.
+///
+/// An ordinary (not in buffer_defaults, per-buffer, or per-keyboard)
+/// lisp variable is actually a field in `struct emacs_globals'.  The
+/// field's name begins with "f_", which is a convention enforced by
+/// these macros.  Each such global has a corresponding #define in
+/// globals.h; the plain name should be used in the code.
+///
+/// E.g., the global "cons_cells_consed" is declared as "int
+/// f_cons_cells_consed" in globals.h, but there is a define:
+///
+///    #define cons_cells_consed globals.f_cons_cells_consed
+///
+/// All C code uses the `cons_cells_consed' name.  This is all done
+/// this way to support indirection for multi-threaded Emacs.
+#[macro_export]
+macro_rules! defvar_lisp {
+    ($fvname:ident, $lname:expr, $value:expr) => {
+        {
+            static mut o_fwd: ::remacs_sys::Lisp_Objfwd = ::remacs_sys::Lisp_Objfwd {
+                ty: ::remacs_sys::Lisp_Fwd_Obj,
+                objvar: ::std::ptr::null_mut()
+            };
+            unsafe { ::remacs_sys::defvar_lisp(&mut o_fwd,
+                                               $lname.as_ptr() as *const i8,
+                                               &mut ::remacs_sys::globals.$fvname);
+            }
+            unsafe { ::remacs_sys::globals.$fvname = $value; }
+        }
+    };
+}
+#[macro_export]
```

Comment:
One, shouldn't there be an empty line between these macros?
two, can we get a comment point to more info on nopro/pro?